test/policies/user_policy_test.rb

```diff
@@ -8,6 +8,23 @@ def setup
     @user_as_department_manager = create :user, :with_profile, role: 'department_manager'
   end
 
+  test 'Destroy: superadmin can delete other user' do
+    assert_permit @user_as_superadmin, @user_as_social_worker, 'destroy?'
+    assert_permit @user_as_superadmin, @user_as_department_manager, 'destroy?'
+    assert_permit @user_as_other_superadmin, @user_as_social_worker, 'destroy?'
```

Comment:
Maybe the check if an other superadmin can delete the entries another superadmin can is not needed. What is important, is that superadmin cannot delete another superadmin or herself. Exactly what you test in the next two tests.